Form/Type/ColorSelectorType.php

```diff
@@ -33,7 +33,7 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        $resolver->setDefaults(array(
+        $defaults = array(
             'choices' => Colors::getAll(),
```

Comment:
> I really did my best to understand here, read it 4 times, smells bad here :smiley: ... Keep not understanding...

What step in particular? I think you should think hard about the goals : 

- having the user be able to upgrade to next **minor** and have everything work
- having the user be able to adopt "the new way" through configuration and get rid of deprecation messages
- having the tests run "the new way" in 90% of the cases, and use the 10% to test that yeah, you do get a deprecation notice when appropriate

> Independent of Symfony version the configureOptions will trigger the deprecated message because Colors::getAll() (if ($isInverted) true) is always called. I did not catch how to prevent this using Symfony 2.8+

I think you should change `Colors::getAll()` to `Colors::getAll($configurableParameterContainer)` in the code. In the test, set this parameter to false and you will be able to prevent this, right?